roles/smash.nix

```diff
@@ -94,25 +94,91 @@ in {
       chown -R nginx:nginx /var/lib/nginx
     '';
   };
+  services.varnish = {
+    enable = true;
+    config = ''
+      vcl 4.1;
+
+      import std;
+
+      backend default {
+        .host = "127.0.0.1";
+        .port = "3100";
+      }
+      acl purge {
+        "localhost";
+        "127.0.0.1";
+      }
+      sub vcl_recv {
+        unset req.http.x-cache;
+        # allow PURGE from localhost and 192.168.55...
+        if (req.method == "PURGE") {
+          if (!client.ip ~ purge) {
+            return(synth(405,"Not allowed."));
+          }
+          return (purge);
+        }
+      }
+      sub vcl_hit {
+        set req.http.x-cache = "hit";
+      }
+      sub vcl_miss {
+        set req.http.x-cache = "miss";
+      }
+
+      sub vcl_pass {
+        set req.http.x-cache = "pass";
+      }
+
+      sub vcl_pipe {
+        set req.http.x-cache = "pipe";
+      }
+
+      sub vcl_synth {
+        set req.http.x-cache = "synth synth";
+        set resp.http.x-cache = req.http.x-cache;
+      }
+
+      sub vcl_deliver {
+        if (obj.uncacheable) {
+          set req.http.x-cache = req.http.x-cache + " uncacheable";
+        } else {
+          set req.http.x-cache = req.http.x-cache + " cached";
+        }
+        set resp.http.x-cache = req.http.x-cache;
+      }
+      sub vcl_backend_response {
+        set beresp.ttl = 30d;
+        if (beresp.status == 404) {
+          set beresp.ttl = 1h;
+        }
+      }
+    '';
+  };
   services.nginx = {
     enable = true;
     package = nginxSmash;
     recommendedGzipSettings = true;
     recommendedOptimisation = true;
     recommendedProxySettings = true;
-    preStart = ''
-      [ -d "${nginxCachePath}" ] || { echo "The nginx data cache dir does not exist"; exit 1; }
-      [ -w "${nginxCachePath}" ] || { echo "The nginx data cache dir is not writable by nginx"; exit 1; }
-    '';
+    #preStart = ''
+    #  [ -d "${nginxCachePath}" ] || { echo "The nginx data cache dir does not exist"; exit 1; }
+    #  [ -w "${nginxCachePath}" ] || { echo "The nginx data cache dir is not writable by nginx"; exit 1; }
+    #'';
```

Comment:
We can just remove this too if the rest of nginx caching is being deleted in favor of varnish.